### PR TITLE
Feat/createtest 3.37

### PR DESCRIPTION
**Pull request by Madhuv2**
create test functionality final version

### CODE REVIEW

File: src/main/java/com/yash/ota/service/TestService.java [hunk 1]
```diff
@@ -111,4 +111,12 @@
 	 */
 	Test getAssignedTestById(int testId) throws InvalidIdException;
 
+	/**
+	 * Method to insert a User into the database
```
Comment (neel136846) [inline src/main/java/com/yash/ota/service/TestService.java]: Change this documentation. It should say test not user.

File: src/main/java/com/yash/ota/service/TopicService.java [hunk 2]
```diff
@@ -42,6 +42,15 @@
 	 * @return TopicRowMapper
 	 */
 	public TopicRowMapper getTopicRowMapper();
+
+    /**
+     * Get the list of topics based on technologyId
+     * @param techId
+     * @return
```
Comment (neel136846) [inline src/main/java/com/yash/ota/service/TopicService.java]: Say what this returns in the documentation

File: src/main/java/com/yash/ota/serviceimpl/TechnologyServiceImpl.java [hunk 3]
```diff
@@ -75,13 +77,30 @@ public TechnologyRowMapper getTechnologyRowMapper() {
 		return this.technologyRowMapper;
 	}
 
+	/**
+	 *
+	 * @return
```
Comment (neel136846) [inline src/main/java/com/yash/ota/serviceimpl/TechnologyServiceImpl.java]: Finish this documentation comment.

File: src/main/java/com/yash/ota/serviceimpl/TestServiceImpl.java [hunk 4]
```diff
@@ -168,7 +168,18 @@ public void setDao(TestDAO dao) {
 		return testDAO.findAll();
 	}
 
-	/**
+    /*
+  get all tests
+  @Author Madhuri Vutukury
```
Comment (neel136846) [inline src/main/java/com/yash/ota/serviceimpl/TestServiceImpl.java]: Fix the formatting of this comment block

File: src/main/java/com/yash/ota/serviceimpl/TopicServiceImpl.java [hunk 5]
```diff
@@ -75,6 +74,18 @@ public Topic getTopicById(int id) {
         return topicDAO.findById(id);
     }
 
+    /**
```
Comment (neel136846) [inline src/main/java/com/yash/ota/serviceimpl/TopicServiceImpl.java]: Add a description for what this method does in the comment block

File: src/test/java/com/yash/ota/daoimpl/TestCreateDAOImplTest.java [hunk 6]
```diff
@@ -0,0 +1,228 @@
+//package com.yash.ota.daoimpl;
```
Comment (neel136846) [inline src/test/java/com/yash/ota/daoimpl/TestCreateDAOImplTest.java]: Do something with this file.

File: src/test/java/com/yash/ota/serviceimpl/TestCreateServiceImplTest.java [hunk 7]
```diff
@@ -0,0 +1,258 @@
+//package com.yash.ota.serviceimpl;
```
Comment (neel136846) [inline src/test/java/com/yash/ota/serviceimpl/TestCreateServiceImplTest.java]: Do something with this

File: src/test/java/com/yash/ota/serviceimpl/TestServiceImplTest.java [hunk 8]
```diff
@@ -165,4 +165,10 @@ public void testGetAssignedTestById_NonExistentTestIdGiven_ShouldThrow() throws
 		List<com.yash.ota.domain.Test> mockedtestInfoList = Arrays.asList(testInfo, testInfo2);
 		return mockedtestInfoList;
 	}
+
```
Comment (neel136846) [inline src/test/java/com/yash/ota/serviceimpl/TestServiceImplTest.java]: Why is this commented out?

File: src/main/java/com/yash/ota/serviceimpl/TopicServiceImpl.java (hunk 5 above)
Comment (karlroth) [reply src/main/java/com/yash/ota/serviceimpl/TopicServiceImpl.java]: Not necessary in Impl

**Pull request merged**